public/components/trace_analytics/components/services/service_view.tsx

```diff
@@ -155,126 +141,74 @@ export function ServiceView(props: ServiceViewProps) {
     else if (setCurrentSelectedService) setCurrentSelectedService(service);
   };
 
-  const redirectToServiceTraces = () => {
-    if (setCurrentSelectedService) setCurrentSelectedService('');
-    setRedirect(true);
-    const filterField =
-      mode === 'data_prepper' || mode === 'custom_data_prepper'
-        ? 'serviceName'
-        : 'process.serviceName';
-    props.addFilter({
-      field: filterField,
-      operator: 'is',
-      value: props.serviceName,
-      inverted: false,
-      disabled: false,
-    });
-
-    const tracesPath = '#/traces';
-    const dataSourceId = props.dataSourceMDSId[0]?.id || '';
-    const urlParts = window.location.href.split('?');
-    const queryParams =
-      urlParts.length > 1 ? new URLSearchParams(urlParts[1]) : new URLSearchParams();
-
-    const modeParam = queryParams.get('mode') || '';
-    const modeQuery = modeParam ? `&mode=${encodeURIComponent(modeParam)}` : '';
-
-    if (newNavigation) {
-      coreRefs.application?.navigateToApp(observabilityTracesNewNavID, {
-        path: `${tracesPath}?datasourceId=${encodeURIComponent(dataSourceId)}${modeQuery}`,
-      });
-    } else {
-      window.location.assign(
-        `${tracesPath}?datasourceId=${encodeURIComponent(dataSourceId)}${modeQuery}`
-      );
-    }
+  const serviceActionsMenu = (isFlyout: boolean) => {
+    return (
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup justifyContent="center" gutterSize="s">
+          <EuiFlexItem
+            grow={false}
+            onClick={() =>
+              redirectToServiceTraces({
```

Comment:
maybe call `setCurrentSelectedService` and `setRedirect` above instead of passing to `redirectToServiceTraces`, since they are only specific to this component

Reply:
Agree, updated it. [3de427e](https://github.com/opensearch-project/dashboards-observability/pull/2398/commits/3de427e9fe08191d804ca0496fec4e718e398b32)